src/pages/Upload.js

```diff
@@ -739,9 +739,21 @@ const UploadData = _.flow( // eslint-disable-line lodash-fp/no-single-compositio
 
   // State
   const { query } = Nav.useRoute()
-  const [currentStep, setCurrentStep] = useState(StateHistory.get().currentStep || 'workspaces')
   const [workspaceId, setWorkspaceId] = useState(query.workspace)
   const [collection, setCollection] = useState(query.collection)
+  const [currentStep, setCurrentStep] = useState(() => {
+    if (StateHistory.get().currentStep) {
+      return StateHistory.get().currentStep
+    }
+    // If workspace and/or collection were provided in the URL, start at a later step
+    if (workspaceId) {
+      if (collection) {
+        return 'data'
+      }
+      return 'collection'
+    }
+    return 'workspaces'
+  })
```

Comment:
`Utils.cond` would help us write this without multiple `return`s, which may make it easier to reason about.

```suggestion
  const [currentStep, setCurrentStep] = useState(Utils.cond(
    [!!StateHistory.get().currentStep, () => StateHistory.get().currentStep],
    [!!workspaceId && !!collection, () => 'data'],
    [!!workspaceId && !collection, () => 'collection'],
    () => 'workspaces'
  ))
```

Note that we removed the laziness (the `() =>` piece) because it didn't seem needed. 

What do you think?